### ISSUE TITLE
new option in filmic display tab

### ISSUE THREAD

**Issue by SoupyGit** [darktable-org/darktable#12850]
**Is your feature request related to a problem? Please describe.**
New users often complain about the difficulty understanding all the different filmic parameters.

**Describe the solution you'd like**
Above target black and white luminance sliders, have a dropdown box called 'target' (you may come up with a better name) with options such as: as per output profile, SDR, HDR, custom, (you may come up with better options), whereby the values of target black and white luminance auto adjust according to the option chosen. 

Default would possibly be ‘as per output profile’, which reads the profile selected in ‘output profile’ module and adjusts the target black and white sliders accordingly. (eg. if sRGB is selected, values would be 0.0152% and 100%. If linear rec 2020 was selected, they could be 0% and 100%. The values corresponding to each profile would need to be decided by someone smarter than I, this just presents the idea of functionality). If 'custom' was chosen, user could input their own values. 

The benefit of this feature is that users would not have to understand what on earth target black and white luminance meant, or what values to use in any given scenario - that would be decided by devs who understand these things - and users would only have to know what their output is. There is also the benefit where if the user changes their output profile, they do not have to change filmic as well, it does so automatically.

**Alternatives**
New users have to read the manual. (Although manual in current state doesn't account for all scenarios. For instance, I have heard Aurelien say Target White Luminance needs to be set to 400% for HDR, but I don't see this in the manual - so no user will know what to do in that scenario unless they follow Aurelien).

**Comment by Nilvus** [comment]
For this one, I don't understand how this will help new users. Having such new option will probably lost more users. And I don't understand the need to have multiple targets. The best way to understand what those targets sliders do is to click on them, going to a side then the other and see the effect on the image.

Working on photos, to better do that, need knowledge (so time) AND visualize what we do on photos and effects on existing settings.

**Comment by SoupyGit** [comment]
> And I don't understand the need to have multiple targets.

From the manual:

> The default parameters should work 99% of the time, the remaining 1% being when you output in linear RGB space (REC709, REC2020) for media handling log-encoded data.

So different settings are intended for different outputs. 0.0152-100% for sRGB. 0-100% for linear output. 0-400% for HDR. You can also use the sliders creatively, which is why to allow a 'custom' choice. These are all different targets, for different use cases.

> The best way to understand what those targets sliders do is to click on them, going to a side then the other and see the effect on the image.

To understand them creatively, yes. But they do not just have a creative purpose, so you cannot understand them completely by doing this. As per the manual:

> The target black luminance parameter sets the ground-level black of the target medium. By default it is set to the minimum non-zero value that can be encoded by the available number of bits in the output color space. Reducing it to zero means that some non-zero luminances will be mapped to 0 in the output, potentially losing some detail in the very darkest parts of the shadows.

You do not get that understanding by just moving sliders. Nor will most users know when a 0 or non 0 value is recommended for any given output format. If you just moved sliders left and right, you might set it at 0 for sRGB output and lose shadow detail without realizing.

> For this one, I don't understand how this will help new users.

You don't understand? Or you disagree? I am happy for you to disagree. But in terms of understanding, I think it helps in two ways:

1) They do not need to understand the meaning/purpose of the sliders, or what value is recommended for how many bits or which output format. All they have to do is select their target output from options like: as per output profile, sRGB, Linear, HDR, Custom. And if the default setting is 'as per output profile', they do not even have to think about it, the best solution is found for them.

2) Lets say we use sRGB as default output profile for exporting jpeg. But sometimes we want to output linear rec 2020 tiff to do more work in Gimp. In this proposal, when changing from sRGB to linear rec 2020 in output color profile module, the change from 0.0152-100% to 0-100% would happen automatically in filmic rgb (when 'as per output profile' option is selected). This automatic change would be very handy, because often I forget to go back into filmic and make the change manually, since it is a tab I don't often use.

Hope that clears up misunderstanding.

**Comment by MStraeten** [comment]
if you are satisfied with an one size fits all aproach then you don't need darktable - there are plenty of simpler tools available.
darktable is designed to fit the requirements of those, that want to have full control - and full control requires an at least basic understanding, what the different slider can be used for. 
If that's too much for a beginner, then using presets copied from endless pixls.us rawprocessing threads might be a first step.
If you have a different target user group in mind, then maybe a fork for a lite version might be the better answer...

**Comment by SoupyGit** [comment]
Can we not paint with such broad strokes, and actually focus on what was put forth. What are its pros and cons?

Nowhere does my idea say to take away the possibility of full control, and suggesting new users copy presents for something completely unrelated to the issue raised is absurd.

**Comment by MStraeten** [comment]
you need different settings in dependence of the image - there's no one size fits all for HDR, SDR, ... 
So it's up to the user to make own experienced- and then the user will be able to define his own presets. 
A lot of learning is done simply by doing following edits done by more experienced image editors. So taking their receipts in some cases as starting point receipts for own edits isn't the worst learning approach. So a beginner can populate his preset library and use them based on analogies in the scene to be edited. Just having a few default completely arbitrary options without any relations to a scene a user can imagine is a far inferior way

**Comment by Nilvus** [comment]
@flannelhead: as you know filmic code, what's your view on that one?

**Comment by flannelhead** [comment]
Thanks for the proposal @SoupyGit. Here are some comments based on what I know (and what I don't :D)

For the black point part - I've discussed a couple of months ago with Aurélien that automatic black point compensation would be nice to have in filmic. That would probably be a checkbox in the "display" tab that would disable the black point slider if enabled, and set the black point based on the output profile. This would essentially implement the "as per output profile" choice that you're proposing. Black point compensation should be also a familiar concept from other image editing software. It would be best if it worked automatically by default.

For the white point part and SDR / HDR, it gets more complicated. I'm not really sure about what the white point should be for rendering material for (10-bit) HDR displays. Is anyone here using darktable with a 10-bit HDR display (such that darktable also works in 10 bits)? We would need someone able to test these things before implementing a preset combobox for SDR / HDR. Does @TurboGit have better knowledge of the current status of 10-bit color in darktable?

**Comment by SoupyGit** [comment]
Thank you @flannelhead 
Black Point compensation would be great, a simpler solution to the one I put forth, covering all workflows but HDR.

In my proposal it would be one click to select Hdr. In yours it would be one slider/keyboard entry to change target white to 400%. So main difference is that in mine they wouldn't need to have knowledge of what to set the slider to, but in yours they would.

But I agree Hdr is currently a very rare use case, and testing would need to be done before implementing the option, which might be hard for devs without the hardware. So I find your proposal of black point compensation a good solution.
